deepchecks/tabular/context.py

```diff
@@ -203,6 +204,39 @@ def __init__(
         if test and not train:
             raise DatasetValidationError('Can\'t initialize context with only test. if you have single dataset, '
                                          'initialize it as train')
+
+        # Handle model_classes
+        if model_classes and len(model_classes) == 0:
+            raise DeepchecksValueError('Received empty model_classes')
+        supported_models_link = doclink(
+            'supported-prediction-format',
+            template='For more information please refer to the Supported Models guide {link}')
+        if model_classes and sorted(model_classes) != model_classes:
+            raise DeepchecksValueError(f'Received unsorted model_classes. {supported_models_link}')
+
+        self._task_type, self._observed_classes, self._model_classes = infer_task_type_and_classes(
+            model, train, test, model_classes)
+
+        # Validate probas shape, and make sure we know about all classes if only probabilities have been passed
+        if y_proba_train is not None and not isinstance(y_proba_train, (Sequence, np.ndarray)):
+            raise ValueError('y_proba_train must be a matrix with n_samples rows and n_classes columns')
+        if y_proba_test is not None and not isinstance(y_proba_test, (Sequence, np.ndarray)):
+            raise ValueError('y_proba_test must be a matrix with n_samples rows and n_classes columns')
+        if y_proba_test is not None and np.array(y_proba_test).shape[1] != np.array(y_proba_train).shape[1]:
+            raise DeepchecksValueError(
+                f'y_proba_test with {len(y_proba_test[0])} columns does not match y_proba_train with'
+                f' {len(y_proba_train[0])} columns.'
+            )
+        proba_shape = len(y_proba_train[0]) if y_proba_train is not None else None
+        if (y_proba_test is not None or y_proba_train is not None) and y_pred_train is None and y_pred_test is None:
+            if self._model_classes is None and len(self._observed_classes) != proba_shape:
```

Comment:
A. We are doing this validation when initalizing scorers, not sure if we need to repeat it here
B. Why if inside if instead of end?

Reply:
A. I didn't know about it (assuming you're talking about line 273 in scorer.py). The only difference is that here it's a specific validation in case model_classes wasn't set, and in scorers.py the validation can be triggered either by it not being set or by it being set incorrectly. 
B. Makes it cleaner, rather than a huge if

Reply:
A. feels redundant, what happens currently is:
the user doesn't pass classes + the model doesn't have `classes_` -> if a scorer is needed we are issuing a warning "could not find model classes, using observed classes" -> if proba doesn't match classes we raise exception